routing/bandwidth.go

```diff
@@ -89,30 +107,71 @@ func (b *bandwidthManager) getBandwidth(cid lnwire.ShortChannelID,
 		return 0
 	}
 
-	// If our link isn't currently in a state where it can  add another
-	// outgoing htlc, treat the link as unusable.
-	if err := link.MayAddOutgoingHtlc(amount); err != nil {
-		log.Warnf("ShortChannelID=%v: cannot add outgoing htlc: %v",
-			cid, err)
+	if len(blobs) == 0 {
+		// If our link isn't currently in a state where it can add
+		// another outgoing htlc, treat the link as unusable.
+		if err := link.MayAddOutgoingHtlc(amount); err != nil {
+			log.Warnf("ShortChannelID=%v: cannot add outgoing "+
+				"htlc: %v", cid, err)
+			return 0
+		}
+
+		// Otherwise, we'll return the current best estimate for the
+		// available bandwidth for the link.
+		return link.Bandwidth()
+	}
+
+	// If the provided tlv blobs are not of the expected number, then we'll
+	// log a warning and return zero bandwidth.
+	if len(blobs) != 2 {
```

Comment:
Can we modify the type signature to more explicitly encode some of the desired semantics?